# Bathycoccus genome assembly and annotation

This repository contains all scripts used to generate de novo whole genome assembly from Bathycoccus sp. strains sequenced with both Oxford Nanopore Technology long reads and Illumina short reads.

Whole genome sequences and basecalled reads of the strains sequenced in this study are available at the European Nucleotide Archive under project accession PRJEB67444.

## Summary

Population-scale sequencing has become a standard practice to explore the natural genetic diversity underlying adaptation, notably in land plants. However, current sequencing initiatives for eukaryotic phytoplankton primarily concentrate on creating reference genomes for model organisms and characterizing natural communities through metagenomics approaches. Consequently, few species have been thoroughly sequenced and intraspecific genetic diversity remains virtually undescribed, limiting our understanding of diversity and adaptation mechanisms. Here we report a biological and genomic resource to explore the genetic diversity of the cosmopolitan and ecologically important Bathycoccus genus. To span broad geographical and temporal scales, we selected available strains in collection and previously isolated strains from the Banyuls bay (Mediterranean sea) as well as newly isolated strains from the Baffin bay (Arctic ocean). By combining Oxford Nanopore Technology long reads and Illumina short reads technologies, we produced and annotated 28 Bathycoccus sp. de novo assembled genomes of high quality, including 24 genomes of Bathycoccus prasinos strains along a latitudinal gradient between 40° and 78° North, one reference genome of the Bathycoccus calidus species and 3 genomes of a yet undescribed Bathycoccus species named Bathycoccus catiminus. We assessed the genetic diversity of this genus through phylogenomic analyses and highlighted the central role of this genomic resource in providing new insights into the diversity of outlier chromosomal structures. The Bathycoccus biological and genomic resources offer a robust framework for investigating the diversity and adaptation mechanisms of eukaryotic phytoplankton in the Ocean.

## Reference

Dennu, L., Devic, M., Rigonato, J., Falciatore, A., Lozano, J.-C., Vergé, V., Mariac, C., Jaillon, O., The DarkEdge Sampling Team, Sabot, F., & Bouget, F.-Y. (2023). Biological and genomic resources for the cosmopolitan phytoplankton Bathycoccus: Insights into genetic diversity and major structural variations. https://doi.org/10.1101/2023.10.16.562038

